doc/developer/design/20220330_persist.md

```diff
@@ -781,20 +843,20 @@ For location errors, on the other hand, we might retry within a given timeout
 window.
 
 Another important axis along which we differentiate errors is _definiteness_
-(TODO(aljoscha): come up with a better term because this one is already doing a
-lot of work elsewhere). _Definite_ errors are those errors where we know that
-some operation definitely failed, while _indefinite_ errors are those errors
-where we cannot know. This distinction is important for write operations (all
-operations that change state): if an operation definitely failed, we can retry
-without running the risk of writing duplicate data. If, on the other hand, we
-get a timeout or some other indefinite errors, we cannot know that the data was
-not written and have to abort the complete write attempt.
-
-TODO(aljoscha): Figure out how to fit the two axes of errors into an ergonomic
-API. Maybe definite/indefinite are variants of `LocationError`? It also feels
-like usage error/location error sometimes map to definite/indefinite, but only
-if you squint hard enough. I also think we will need to play this by ear, see
-what’s ergonomic when we start implementing things on top.
+(TODO: Come up with a better term because this one is already doing a lot of
+work elsewhere). _Definite_ errors are those errors where we know that some
```

Comment:
spitballing - it kind of feels like Rust's default Ok/Error split makes things extra convoluted here. Really the top level split is:

- did we get a `Known` / `Unknown` response for this request - where `Known` responses mean that we know exactly what the system did with this request (kyle's version of definite), and `Unknown` responses mean we don't exactly know what the system did with this request.

And then within `Known` there's a split of `Ok`/`Error`, whereas `Unknown` are always some other flavor of errors that are disjoint from the errors possible within `Known`.


This is just a spitball feel free to tioli